client/games/zombies/zombies.py

```diff
@@ -0,0 +1,362 @@
+  def update(self, t, heros, npcs, occupied):
+    pass
+  def moveToward(self, x, y, occupied):
+    newx = wrap(self.x + sign(x-self.x),WIDTH)
+    newy = wrap(self.y + sign(y-self.y),HEIGHT)
+    if occupied:
+      if occupied[newy][newx]==0:
+        occupied[self.y][self.x] = 0
+        self.x = newx
+        self.y = newy
+        occupied[self.y][self.x] = 1
+    else:
+      self.x = newx
+      self.y = newy      
+
+
+class Hero(Character):
+  def __init__(self):
+    self.x       = WIDTH//2
+    self.y       = HEIGHT//2
+    self.color   = (0,255,0)
+    self.type    = 'hero'
+    self.dx      = 0
+    self.dy      = 0
+    self.last_dx = 0
+    self.last_dy = 0
+    self.speed   = 3
+    self.ammo    = 40
+    self.alive   = 40
+  def update(self, t, heros, npcs, occupied):
+    if t%self.speed==0:
+      self.moveToward(self.x+self.dx, self.y+self.dy, None)
+
+
+class Zombie(Character):
+  def __init__(self):
+    self.x     = 0
+    self.y     = 0
+    self.color = (255,0,0)
+    self.type  = 'zombie'
+    self.speed = 6
+    self.alive = 1
+
+    direc  = random.randint(0,4)
+    if direc==0: #East
+      self.y = random.randint(0,HEIGHT-1)
+    elif direc==1: #North
+      self.x = random.randint(0,WIDTH-1)
+    elif direc==2: #West
+      self.y = random.randint(0,HEIGHT-1)
+      self.x = WIDTH-1
+    elif direc==3: #South
+      self.x = random.randint(0,WIDTH-1)
+      self.y = HEIGHT-1
+
+  def update(self, t, heros, npcs, occupied):
+    if t%self.speed==0:
+      self.moveToward(heros[0].x, heros[0].y, occupied)
+    if self.x==heros[0].x and self.y==heros[0].y:
+      heros[0].alive          -= 1
+      self.alive               = 0
+      occupied[self.y][self.x] = 0
+
+class Bullet(Character):
+  def __init__(self, x, y, dx, dy):
+    self.x     = x
+    self.y     = y
+    self.color = (0,0,255)
+    self.type  = 'bullet'
+    self.speed = 1
+    self.dx    = dx
+    self.dy    = dy
+    self.alive = 10
+
+  def update(self, t, heros, npcs, occupied):
+    self.alive = self.alive-1
+    if t%self.speed==0:
+      self.moveToward(self.x+self.dx, self.y+self.dy, None)
+    if occupied[self.y][self.x]!=0:
+      for n in npcs:
+        if n.x==self.x and n.y==self.y:
+          n.alive = 0
+          occupied[self.y][self.x] = 0
+
+class Ammo(Character):
+  def __init__(self):
+    self.x = random.randint(0,WIDTH-1)
+    self.y = random.randint(0,HEIGHT-1)
+    self.color = (0,0,255)
+    self.type  = 'ammo'
+    self.alive = 1
+    self.colors = [(255,255,0), (255,0,255), (0,255,255)]
+
+  def update(self, t, heros, npcs, occupied):
+    h=heros[0]
+    if h.x==self.x and h.y==self.y:
+      h.ammo = min(h.ammo + 20,40)
+      self.alive = 0
+      return
+    self.color = self.colors[t%3]
+
+class Game:
+  def __init__(self):
+    self.screen = Flaschen(host='ft.noise', port=1337, width=WIDTH, height=HEIGHT, layer=5, transparent=False)
+    self.npcs   = []
+    self.heros  = [Hero()]
+    self.main   = Tk()
```

Comment:
Is there an advantage to use a graphical window to receive key-presses vs. just simply from the terminal directly ?
It took me a while to realize that there was a tiny window popping up that I have to navigate the cursor in to even receive keypresses. It would be more natural to just have this as input on the terminal you invoke it.
(besides easier matching expectations, it is less dependencies, easier to play remotely etc.).

WDYT ?